backend/src/game/cards/deck_handler.py

```diff
@@ -0,0 +1,71 @@
+from typing import List
+
+import backend.src.game.cards.definitions as definitions
+from backend.src.game.cards.card import Card
+from backend.src.game.cards.deck import Deck
+
+
+class DeckHandler:
+    """
+    DeckHandler manages the draw, discard, and store pile interactions.
+
+    Use this to purchase power cards from the store and discard used cards.
+    creating a DeckHandler object automatically populates the draw pile from the kot_cards.json.
+    The draw pile is automatically re-populated with discarded cards upon running out.
+    """
+    __draw_pile: Deck = Deck()
+    __discard_pile: Deck = Deck()
+    __card_store: List[Card] = list()
+
+    def __init__(self):
+        self.store.clear()
+        self.discard_pile.clear()
+        self.__draw_pile.get_new_deck()
+        self.__draw_pile.shuffle()
+        self.__fill_card_store()
+
+    def __len__(self):
+        return len(self.draw_pile) + len(self.store) + len(self.discard_pile)
+
+    @property
+    def draw_pile(self):
+        return self.__draw_pile
+
+    @property
+    def store(self):
+        return self.__card_store
+
+    @property
+    def discard_pile(self):
+        return self.__discard_pile
+
+    def get_top_draw_pile_card(self):
```

Comment:
Tests for each of the possible branches in this method, please

Reply:
The tests are already in deck_handler_test.py

normal ->test_buy_single_card
if draw pile is empty -> test_auto_reshuffle
if discard is also empty-> test_run_out_of_cards